cmd/tckconvert.cpp

```diff
@@ -190,6 +200,294 @@ class ASCIIWriter: public WriterInterface<float>
+                vertexOF << (int)( 255 * fabs(T[0])) << " " << (int)( 255 * fabs(T[1])) << " " << (int)( 255 * fabs(T[2])) << "\n";
+                if ( !isLast ) {
+                    faceOF << "3"
+                           << " " << num_vertices + faces(sideIdx,0)
+                           << " " << num_vertices + faces(sideIdx,1)
+                           << " " << num_vertices + faces(sideIdx,2) << "\n";
+                    faceOF << "3"
+                           << " " << num_vertices + faces(sideIdx,3)
+                           << " " << num_vertices + faces(sideIdx,4)
+                           << " " << num_vertices + faces(sideIdx,5) << "\n";
+                    num_faces += 2;
+                }
+            }
+            // Cap the first point, remebering the right hand rule
+            if ( isFirst ) {
+                for ( auto sideIdx = nSides - 1; sideIdx >= 2; --sideIdx ) {
+                    faceOF << "3"
+                           << " " << num_vertices + sideIdx
+                           << " " << num_vertices + sideIdx - 1
+                           << " " << num_vertices << "\n";
+                }
+                num_faces += nSides - 2;
+                isFirst = false;
+            }
+            if ( isLast ) {
+                // faceOF << "Writing end cap, num_vertices = " << num_vertices << "\n";
+                for ( auto sideIdx = 2; sideIdx <= nSides - 1; ++sideIdx ) {
+                    faceOF << "3"
+                           << " " << num_vertices
+                           << " " << num_vertices + sideIdx
+                           << " " << num_vertices + sideIdx - 1 << "\n";
+                }
+                num_faces += nSides - 2;
+            }
+            // We needed to maintain the number of vertices for the caps, now increment for the "circles"
+            num_vertices += nSides;
+        }
+        return true;
+    }
+
+    ~PLYWriter() {
+        // write out list of tracks:
+        vertexOF.close();
+        faceOF.close();
+
+        out <<
+            "ply\n"
+            "format ascii 1.0\n"
+            "comment written by tckconvert v" << mrtrix_version << "\n"
+            "comment part of the mtrix3 suite of tools (http://www.mrtrix.org/)\n"
+            "element vertex " << num_vertices << "\n"
+            "property float32 x\n"
+            "property float32 y\n"
+            "property float32 z\n"
+            "property uchar red\n"
+            "property uchar green\n"
+            "property uchar blue\n"
+            "element face " << num_faces << "\n"
+            "property list uint8 int32 vertex_indices\n"
+            "end_header\n";
+
+        std::ifstream vertexIF(vertexFilename);
+        out << vertexIF.rdbuf();
+        vertexIF.close();
+        std::ifstream faceIF(faceFilename);
+        out << faceIF.rdbuf();
+        faceIF.close();
+        std::remove ( vertexFilename.c_str() );
+        std::remove ( faceFilename.c_str() );
```

Comment:
I'd probably use the _MRtrix3_ [File::unlink()](https://github.com/MRtrix3/mrtrix3/blob/master/lib/file/utils.h#L185-L189) call here, but it's not a big deal - this will work just as well.